src/lib/tsc.ts

```diff
@@ -334,3 +334,47 @@ export function getAbsoluteImportPath(sourceFile: TSM.SourceFile) {
 
   return { isNode, modulePath }
 }
+
+/**
+ * Find the modules in the project that import the given dep and return info about how that dep is imported along with sourceFile ref itself.
+ */
+export function findModulesThatImportModule(project: TSM.Project, id: string) {
+  const data: ModulesWithImportSearchResult[] = []
+
+  for (const sourceFile of project.getSourceFiles()) {
+    if (sourceFile.getFilePath().includes('node_modules')) continue
+    let entry: ModulesWithImportSearchResult
+    for (const importDec of sourceFile.getImportDeclarations()) {
+      if (importDec.getModuleSpecifier().getText() === `'${id}'`) {
```

Comment:
A bonus of this new approach is that we'll have better perf. We used to analyze all source files, even if they didn't import `nexus`. Not sure how many real-world cases will see improvement, but for some larger apps I could see this being a not-nothing improvement.